_posts/2025-02-20-agentic-ai-with-quarkus-p2.adoc

```diff
@@ -0,0 +1,312 @@
+---
+layout: post
+title: 'Agentic AI with Quarkus - part 2'
+date: 2025-02-20T00:00:00Z
+tags: ai llm agents
+synopsis: 'Agentic AI with Quarkus - part 2'
+author: mariofusco
+---
+:imagesdir: /assets/images/posts/agentic
+
+The https://quarkus.io/blog/agentic-ai-with-quarkus/[first part] of this blog post gave a quick introduction on agentic AI and discussed the workflow patterns in which the coordination of the multiple AI services in use is defined programmatically through predefined code paths. In this second part we will explore how agents works and use them in practice with a set of examples.
+
+== Agents
+
+Agents differ from the workflow patterns because the control flow, instead of being implemented programmatically, is entirely delegated to LLMs, which typically invoke tools to complete determinate subtasks. These tools can be seen as functions, opportunistically invoked by the LLM to achieve its goal and implemented in various technologies, like simple imperative methods, external web services, or even other LLMs.
+
+=== The weather forecast agent
+
+This https://github.com/mariofusco/quarkus-agentic-ai/blob/main/src/main/java/org/agenticai/aiastool[first example] of agentic AI implements the following https://github.com/mariofusco/quarkus-agentic-ai/blob/main/src/main/java/org/agenticai/aiastool/WeatherForecastAgent.java[weather forecast agent]
```

Comment:
would be good to introduce this section with a sentence about what your weather forecast agent is expected to accomplish.